sklearn/cluster/_bisect_k_means.py

```diff
@@ -0,0 +1,1215 @@
+    ----------
+    n_clusters : int, default=8
+        The number of clusters to form as well as the number of
+        centroids to generate.
+
+    init : {'k-means++', 'random'} or callable, default='k-means++'
+        Method for initialization:
+
+        'k-means++' : selects initial cluster centers for k-mean
+        clustering in a smart way to speed up convergence. See section
+        Notes in k_init for more details.
+
+        'random': choose `n_clusters` observations (rows) at random from data
+        for the initial centroids.
+
+        If a callable is passed, it should take arguments X, n_clusters and a
+        random state and return an initialization.
+
+    n_init : int, default=10
+        Number of time the inner k-means algorithm will be run with different
+        centroid seeds in each bisection.
+        That will result producing for each bisection best output of n_init
+        consecutive runs in terms of inertia.
+
+    random_state : int, RandomState instance or None, default=None
+        Determines random number generation for centroid initialization. Use
+        an int to make the randomness deterministic.
+        See :term:`Glossary <random_state>`.
+
+    max_iter : int, default=30
+        Maximum number of iterations of the inner k-means algorithm at each
+        bisection.
+
+    verbose : int, default=0
+        Verbosity mode.
+
+    tol : float, default=1e-4
+        Relative tolerance with regards to Frobenius norm of the difference
+        in the cluster centers of two consecutive iterations  to declare
+        convergence. Used in inner k-means algorithm at each bisection to pick
+        best possible clusters.
+
+    copy_x : bool, default=True
+        When pre-computing distances it is more numerically accurate to center
+        the data first. If copy_x is True (default), then the original data is
+        not modified. If False, the original data is modified, and put back
+        before the function returns, but small numerical differences may be
+        introduced by subtracting and then adding the data mean. Note that if
+        the original data is not C-contiguous, a copy will be made even if
+        copy_x is False. If the original data is sparse, but not in CSR format,
+        a copy will be made even if copy_x is False.
+
+    algorithm : {"lloyd", "elkan", "auto", "full"}, default="lloyd"
+        K-means algorithm to use. The classical EM-style algorithm is `"lloyd"`.
+        The `"elkan"` variation can be more efficient on some datasets with
+        well-defined clusters, by using the triangle inequality. However it's
+        more memory intensive due to the allocation of an extra array of shape
+        `(n_samples, n_clusters)`.
+
+        `"auto"` and `"full"` are deprecated and they will be removed in
+        Scikit-Learn 1.3. They are both aliases for `"lloyd"`.
+
+    bisecting_strategy : {"biggest_inertia", "largest_cluster"}, default="biggest_inertia"
```

Comment:
here is the last lint error
```
sklearn/cluster/_bisect_k_means.py:827:89: E501 line too long (90 > 88 characters)
```

Reply:
To be honest - I've get confused and don't know how to multiline it properly

Reply:
I think this should do it...

```suggestion
    bisecting_strategy : {"biggest_inertia", "largest_cluster"},
            default="biggest_inertia"
```

Reply:
@michalkrawczyk I just realize that this lint error is in a `BisectingKMeans2` class, never imported and similar to the `BisectingKmeans` function. You are modifying the docstring of `BisectingKmeans` instead and the error is not fixed.
Is `BisectingKMeans2` needed?

Reply:
`BisectingKMeans2` is an alternative implementation to `BisectingKMeans` that is currently being tested. The docstring can be sorted out once the best implementation is selected.

Reply:
Thanks @TomDLT ! The only problem is that the linting error blocks all the checks but the "Linux_Runs pylatest_conda_forge_mkl". Perhaps it is worth to be fixed in order to verify if the implementation is ok?